includes/blocks/class-sensei-lesson-blocks.php

```diff
@@ -76,15 +76,30 @@ public function initialize_blocks() {
 
 		$post_type_object = get_post_type_object( 'lesson' );
 
-		$block_template = [
-			[ 'sensei-lms/lesson-properties' ],
-			[ 'sensei-lms/button-contact-teacher' ],
-			[
-				'core/paragraph',
-				[ 'placeholder' => __( 'Write lesson content...', 'sensei-lms' ) ],
-			],
-			[ 'sensei-lms/lesson-actions' ],
-		];
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$lesson_id            = isset( $_GET['post'] ) ? intval( $_GET['post'] ) : 0;
+		$course_id            = Sensei()->lesson->get_course_id( $lesson_id );
```

Comment:
Can't you use `Sensei_Utils::get_current_course()` here?

Also I am a little concerned about the `$_GET['post']` not existing or not being a number and using a 0 as lesson id. Maybe we could detect that and do an early return.

Reply:
Thank you for the review, Aarón!

> Can't you use Sensei_Utils::get_current_course() here?

Unfortunately, in that case, it doesn't work because the moment (hook) we call the `initialize_blocks`. At that moment, the query is not ready yet.

> Also I am a little concerned about the $_GET['post'] not existing or not being a number and using a 0 as lesson id. Maybe we could detect that and do an early return.

That will happen to new courses, and it's expected because new courses won't have associated courses yet. When a course is created through the Course Outline it would work well.

Reply:
I added some comments to make it documented in the code: https://github.com/Automattic/sensei/pull/4558/commits/a303e2af11f570e4b8c3177584102965cb6bd0a1
Let me know if you have a different idea for that. ;)

Reply:
Understood! Thanks for the detailed explanation :)

Small notice: we will need to take this into account when implementing the global toggle override for enabling the sensei theme.